aws_helper/config.go

```diff
@@ -22,5 +25,29 @@ func CreateAwsSession(awsRegion, awsProfile string) (*session.Session, error) {
 		return nil, errors.WithStackTraceAndPrefix(err, "Error finding AWS credentials (did you set the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables?)")
 	}
 
+	if awsRoleArn != "" {
+		err = assumeRole(session, awsRoleArn)
```

Comment:
This method calls `Credentials.Get()` a few lines above, so you should probably handle the role stuff earlier. Moreover, I think it can be done simpler, without the need for an extra method:

```go
func CreateAwsSession(awsRegion, awsProfile, awsRoleArn string) (*session.Session, error) {
  session, err := // ...

  if awsRoleArn != "" {
    session.Config.Credentials = stscreds.NewCredentials(sess, assumeRoleArn)
  }

  _, err = session.Config.Credentials.Get()
  // ...
```